src/sdam/server_selection_events.ts

```diff
@@ -0,0 +1,137 @@
+import {
+  SERVER_SELECTION_FAILED,
+  SERVER_SELECTION_STARTED,
+  SERVER_SELECTION_SUCCEEDED,
+  WAITING_FOR_SUITABLE_SERVER
+} from '../constants';
+import { type ReadPreference } from '../read_preference';
+import { type ServerSelector } from './server_selection';
+import type { TopologyDescription } from './topology_description';
+
+/**
+ * The base export class for all logs published from server selection
+ * @internal
+ * @category Log Type
+ */
+export abstract class ServerSelectionEvent {
```

Comment:
I have noticed that bulk operations use `operationId`. Is it a different id? Or we should add operationId as an optional prop of the ServerSelectionEvent class in case it presents.